mongodbatlas/resource_mongodbatlas_advanced_cluster_migration_test.go

```diff
@@ -1,132 +1,135 @@
-			map[string]interface{}{
-				"enabled":         1,
-				"read_preference": "secondary",
-			},
-		},
-	}
-
-	v0Config := terraform.NewResourceConfigRaw(v0State)
-	diags := resourceMongoDBAtlasAdvancedClusterResourceV0().Validate(v0Config)
-
-	if len(diags) > 0 {
-		t.Error("test precondition failed - invalid mongodb cluster v0 config")
-
-		return
-	}
-
-	// test migrate function
-	v1State := migrateBIConnectorConfig(v0State)
-
-	v1Config := terraform.NewResourceConfigRaw(v1State)
-	diags = resourceMongoDBAtlasAdvancedCluster().Validate(v1Config)
-	if len(diags) > 0 {
-		fmt.Println(diags)
-		t.Error("migrated cluster advanced config is invalid")
-
-		return
-	}
-}
-
-func TestAccClusterRSAdvancedClusterV0StateUpgrade_ReplicationSpecs(t *testing.T) {
-	v0State := map[string]interface{}{
-		"project_id":     "test-id",
-		"name":           "test-cluster",
-		"cluster_type":   "REPLICASET",
-		"backup_enabled": true,
-		"disk_size_gb":   256,
-		"replication_specs": []interface{}{
-			map[string]interface{}{
-				"zone_name": "Test Zone",
-				"region_configs": []interface{}{
-					map[string]interface{}{
-						"priority":      7,
-						"provider_name": "AWS",
-						"region_name":   "US_EAST_1",
-						"electable_specs": []interface{}{
-							map[string]interface{}{
-								"instance_size": "M30",
-								"node_count":    3,
-							},
-						},
-						"read_only_specs": []interface{}{
-							map[string]interface{}{
-								"disk_iops":     0,
-								"instance_size": "M30",
-								"node_count":    0,
-							},
-						},
-						"auto_scaling": []interface{}{
-							map[string]interface{}{
-								"compute_enabled":            true,
-								"compute_max_instance_size":  "M60",
-								"compute_min_instance_size":  "M30",
-								"compute_scale_down_enabled": true,
-								"disk_gb_enabled":            false,
-							},
-						},
-					},
-				},
-			},
-		},
-	}
-
-	v0Config := terraform.NewResourceConfigRaw(v0State)
-	diags := resourceMongoDBAtlasAdvancedClusterResourceV0().Validate(v0Config)
-
-	if len(diags) > 0 {
-		fmt.Println(diags)
-		t.Error("test precondition failed - invalid mongodb cluster v0 config")
-
-		return
-	}
-
-	// test migrate function
-	v1State := migrateBIConnectorConfig(v0State)
-
-	v1Config := terraform.NewResourceConfigRaw(v1State)
-	diags = resourceMongoDBAtlasAdvancedCluster().Validate(v1Config)
-	if len(diags) > 0 {
-		fmt.Println(diags)
-		t.Error("migrated advanced cluster replication_specs invalid")
-
-		return
-	}
-
-	if len(v1State["replication_specs"].([]interface{})) != len(v0State["replication_specs"].([]interface{})) {
-		t.Error("migrated replication specs did not contain the same number of elements")
-
-		return
-	}
-}
+// import (
+// 	"fmt"
+// 	"testing"
+
+// 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+// )
+
+// This resource cannot migrate to "github.com/hashicorp/terraform-plugin-sdk/terraform"
+// until the related TF resource is migrated to the new framework
```

Comment:
FYI

Reply:
Thanks @andreaangiolillo , could you share the discussion with Terraform? Why does this need to be migrated first? Any dependencies that are stopping us?

Reply:
https://hashicorp.slack.com/archives/CSAQRMZGT/p1688990544295959

Reply:
After updating the import with [github.com/hashicorp/terraform-plugin-testing/terraform](http://github.com/hashicorp/terraform-plugin-testing/terraform) , the test is complaining that the SDKv2Resource.Validate(Config) is expecting a resource under the v2 package. This doc page [Migrating from SDK](https://developer.hashicorp.com/terraform/plugin/testing/migrating#migration-steps) does not provide any suggestions on which function to use.